core/arch/arm/sm/sm_a32.S

```diff
@@ -139,9 +139,23 @@ UNWIND(	.cantunwind)
 	orr	r0, r0, #(SCR_NS | SCR_FIQ) /* Set NS and FIQ bit in SCR */
 	write_scr r0
 
+	dsb
+	isb
+
 	b	.smc_exit
 
 .smc_ret_to_sec:
+	/*
```

Comment:
This is just overhead for platforms not needing it.
It also makes it harder to make future changes to the secure monitor as the TI sm_platform_handler depends on internal stuff of the secure monitor.